db/db.go

```diff
@@ -109,11 +110,24 @@ func NewMinio(env e2e.Environment, name, bktName string, opts ...Option) *e2emon
 		e2e.StartOptions{
 			Image: o.image,
 			// Create the required bucket before starting minio.
-			Command: e2e.NewCommandWithoutEntrypoint("sh", "-c", command+fmt.Sprintf(
-				"su - me -s /bin/sh -c 'mkdir -p %s && %s /opt/bin/minio server --address :%v --quiet %v'",
-				filepath.Join(f.Dir(), bktName), strings.Join(envVars, " "), ports[AccessPortName], f.Dir()),
+			Command: e2e.NewCommandWithoutEntrypoint(
+				"sh",
+				"-c",
+				command+fmt.Sprintf(
+					"su - me -s /bin/sh -c 'mkdir -p %s && %s /opt/bin/minio server --address :%v --console-address :%v --quiet %v'",
+					filepath.Join(f.Dir(), bktName),
+					strings.Join(envVars, " "),
+					ports[AccessPortName],
+					ports[consolePortName],
+					f.Dir(),
+				),
+			),
+			Readiness: e2e.NewHTTPReadinessProbe(
+				consolePortName,
+				"/",
+				200,
+				200,
 			),
-			Readiness: e2e.NewHTTPReadinessProbe(AccessPortName, "/minio/health/live", 200, 200),
```

Comment:
This seems to be the liveness probe and not the readiness probe https://github.com/minio/minio/blob/master/docs/metrics/healthcheck/README.md#readiness-probe.

Maybe we can try switching to `minio/health/ready` and see if that works well? That way we can avoid having to load the full console. 🙂

Reply:
We won't load the "full" console. As Javascript doesn't run, it's much more lightweight. It's also hacky, but it is what works 100% of the time.

Reply:
In fact, MinIO's liveness and readiness check are exactly the same:

https://github.com/minio/minio/blob/444ff20bc5268d61b329c1d1bc7eb22aeb5f96de/cmd/healthcheck-handler.go#L98

Reply:
Oh wow! Ideally, this should be fixed on minio side. 
But since we already have working "console"-based one then I think this is fine then. 

Do you mind opening an issue on minio repo and linking here as a TODO, so that if this is ever fixed, we can update and remove hack here? 🙂

Reply:
I'll try to carve some time to create a snippet to reliably reproduce the issue and start a discussion over there. I prefer to not tie the creation of the issue there to merging work here, especially because their readiness probe "does what it says it does". It'll be more of a design question for them.

Reply:
Yeah, seems like they closed issue too. But IMO, readiness and liveness should mean diff things. Anyway, thanks for adding comment!